index.js

```diff
@@ -173,22 +170,30 @@ Compiler.prototype = {
    * @api public
    */
 
-  buffer: function (str, interpolate) {
+  buffer: function (str, interpolate, node) {
     var self = this;
     if (interpolate) {
       var match = /(\\)?([#!]){((?:.|\n)*)$/.exec(str);
       if (match) {
-        this.buffer(str.substr(0, match.index), false);
+        this.buffer(str.substr(0, match.index), false, node);
         if (match[1]) { // escape
-          this.buffer(match[2] + '{', false);
-          this.buffer(match[3], true);
+          this.buffer(match[2] + '{', false, node);
+          this.buffer(match[3], true, node);
           return;
         } else {
           var rest = match[3];
-          var range = parseJSExpression(rest);
+          var range;
+          try {
+            range = parseJSExpression(rest);
+          } catch (ex) {
+            if (/The end of the string was reached with no closing bracket found\./.test(ex.message)) {
```

Comment:
I'm always worried about regexing for specific error messages. They are not part of a semver contract, they might change at any time and I don't think we have a test case in jade-code-gen that would notice that your fix has stopped working.
Could we instead update character-parser to send an errorCode (part of its semver api) as part of its error object?